packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1339,6 +1339,21 @@ class _TimePickerInputState extends State<_TimePickerInput> with RestorationMixi
   late final RestorableTimeOfDay _selectedTime = RestorableTimeOfDay(widget.initialSelectedTime);
   final RestorableBool hourHasError = RestorableBool(false);
   final RestorableBool minuteHasError = RestorableBool(false);
+  late FocusNode _hourFocusNode;
+  late FocusNode _minuteFocusNode;
+
+  @override
+  void initState() {
+    super.initState();
+    _hourFocusNode = FocusNode()..addListener(() {
+      setState(() {
+      });
+    });
+    _minuteFocusNode = FocusNode()..addListener(() {
+      setState(() {
+      });
+    });
```

Comment:
Why are we calling `setState` when the focus node changes? The fields themselves should automatically rebuild as necessary.